ubcpi/ubcpi.py

```diff
@@ -0,0 +1,542 @@
+"""A Peer Instruction tool for edX by the University of British Columbia."""
+import os
+import random
+from copy import deepcopy
+import uuid
+
+from django.core.exceptions import PermissionDenied
+import pkg_resources
+from webob import Response
+from xblock.core import XBlock
+from xblock.exceptions import JsonHandlerError
+from xblock.fields import Scope, String, List, Dict, Integer
+from xblock.fragment import Fragment
+
+from answer_pool import offer_answer, validate_seeded_answers, get_other_answers
+import persistence as sas_api
+from serialize import parse_from_xml
+
+STATUS_NEW = 0
+STATUS_ANSWERED = 1
+STATUS_REVISED = 2
+
+# this is a limitation in the database submissions_submission.raw_answer field, which is a TEXT and has limit
+# of 64k in size. Because we are storing rationale and revised rationale both in the the field, the max size
+# for the rationale is half
+MAX_RATIONALE_SIZE = 32000
```

Comment:
The answer pool has a size of 50 and each submission has a max size of 64k. @nedbat do you see any concern with storing this in the block? (the rationales are stored as a list of strings, I believe)